sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdContextRequestEncoder.java

```diff
@@ -9,7 +9,7 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-final class RntbdContextRequestEncoder extends MessageToByteEncoder {
+final class RntbdContextRequestEncoder extends MessageToByteEncoder<RntbdContextRequest> {
```

Comment:
@David-Noble-at-work can you please look through some of these rntbd changes ?

Reply:
Adding one note: Netty first checks whether this encoder can handle an object by passing the object to `acceptOutboundMessage` if it returns true then Netty calls `encode` with that object. i.e. encode is called only if the object is of `RntbdContextRequest`.